lib/errors.js

```diff
@@ -109,7 +108,8 @@ exports.Err = class {
 
         return format.replace(/\{\{(\!?)([^}]+)\}\}/g, ($0, isSecure, name) => {
 
-            const value = Hoek.reach(this.context, name);
+            const context = Object.assign({}, this.context, { key: this.context.label });
```

Comment:
@Marsup The context's key wasn't changed, the key was the label before, look at https://github.com/hapijs/joi/pull/1168/files#diff-11c8ed6891535db0efdc438909405929L78

```js
this.context.key = this.flags.label;
```

Before my changes the template language accepted `{key: 'label'}`, which is incorrect, now it accepts `{key: 'label', label: 'label'}`, which is also incorrect, but it is not more incorrect than it was.

I agree that all bugs should be fixed, but if fixing all at once is problematic, we should fix one bug at a time.